file_store/src/verified_subscriber_mapping_event.rs

```diff
@@ -0,0 +1,70 @@
+use crate::{
+    traits::{MsgDecode, MsgTimestamp, TimestampDecode, TimestampEncode},
+    Error, Result,
+};
+use chrono::{DateTime, Utc};
+use helium_crypto::PublicKeyBinary;
+use helium_proto::services::poc_mobile::VerifiedSubscriberMappingEventReqV1;
+use serde::{Deserialize, Serialize};
+use sqlx::Row;
+
+#[derive(Clone, Deserialize, Serialize, Debug, PartialEq)]
+pub struct VerifiedSubscriberMappingEvent {
+    pub subscriber_id: Vec<u8>,
+    pub total_reward_points: u64,
+    pub timestamp: DateTime<Utc>,
+    pub verification_mapping_pubkey: PublicKeyBinary,
+}
+
+impl MsgDecode for VerifiedSubscriberMappingEvent {
+    type Msg = VerifiedSubscriberMappingEventReqV1;
+}
+
+impl MsgTimestamp<Result<DateTime<Utc>>> for VerifiedSubscriberMappingEventReqV1 {
+    fn timestamp(&self) -> Result<DateTime<Utc>> {
+        self.timestamp.to_timestamp()
+    }
+}
+
+impl MsgTimestamp<u64> for VerifiedSubscriberMappingEvent {
+    fn timestamp(&self) -> u64 {
+        self.timestamp.encode_timestamp()
+    }
+}
+
+impl From<VerifiedSubscriberMappingEvent> for VerifiedSubscriberMappingEventReqV1 {
+    fn from(v: VerifiedSubscriberMappingEvent) -> Self {
+        let timestamp = v.timestamp();
+        VerifiedSubscriberMappingEventReqV1 {
+            subscriber_id: v.subscriber_id,
+            total_reward_points: v.total_reward_points,
+            timestamp,
+            verification_mapping_pubkey: v.verification_mapping_pubkey.into(),
+            signature: vec![],
+        }
+    }
+}
+
+impl TryFrom<VerifiedSubscriberMappingEventReqV1> for VerifiedSubscriberMappingEvent {
+    type Error = Error;
+    fn try_from(v: VerifiedSubscriberMappingEventReqV1) -> Result<Self> {
+        let timestamp = v.timestamp()?;
+        Ok(Self {
+            subscriber_id: v.subscriber_id,
+            total_reward_points: v.total_reward_points,
+            timestamp,
+            verification_mapping_pubkey: v.verification_mapping_pubkey.into(),
+        })
+    }
+}
+
+impl sqlx::FromRow<'_, sqlx::postgres::PgRow> for VerifiedSubscriberMappingEvent {
```

Comment:
for better or worse we have tended not to include db/sqlx requirements in the filestore.  Sql support in filestore is an optional feature and thus baking these may lead to issues

Reply:
removed